File: Pattern/Overlapping Intervals/435. Non-overlapping Intervals/Solution.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        
        Arrays.sort(intervals, (a,b)  ->  Integer.compare(a[1],b[1]));

        int[] prev = intervals[0];
        int answer = 0;
        for( int i = 1 ; i < intervals.length ; i++){
            if(intervals[i][0] < prev[1]){
                answer++;
            }
            else{
                prev = intervals[i];
            }
        }

        return answer;
    }
}

File: pattern/Overlapping Intervals/435. Non-overlapping Intervals/Solution.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x:x[1])
        answer = 0
        prev = intervals[0]
        for i in range (1,len(intervals)):
            if intervals[i][0] < prev[1]:
                answer = answer + 1
            else :
                prev = intervals[i]
        return answer